src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/CertificateEncoderDecoder/CertificateEncoderDecoderToolViewModel.cs

```diff
@@ -0,0 +1,299 @@
+            {
+                throw new NotSupportedException();
+            }
+            
+            await SetInputValueAsync(output);
+        }
+
+        private void QueueConversionCalculation()
+        {
+            _conversionQueue.Enqueue((InputValue ?? string.Empty, PasswordValue));
+            TreatQueueAsync().Forget();
+        }
+
+        private async Task TreatQueueAsync()
+        {
+            if (_conversionInProgress)
+            {
+                return;
+            }
+
+            _conversionInProgress = true;
+
+            await TaskScheduler.Default;
+
+            while (_conversionQueue.TryDequeue(out (string? cert, string? password) certAndPw))
+            {
+                string conversionResult = string.IsNullOrEmpty(certAndPw.cert)
+                    ? ""
+                    : await DecodeCertificateDataAsync(certAndPw.cert!, certAndPw.password).ConfigureAwait(false);
+
+                ThreadHelper.RunOnUIThreadAsync(Core.Threading.ThreadPriority.Low, () =>
+                {
+                    OutputValue = conversionResult;
+
+                    if (!_toolSuccessfullyWorked)
+                    {
+                        _toolSuccessfullyWorked = true;
+                        _marketingService.NotifyToolSuccessfullyWorked();
+                    }
+                }).ForgetSafely();
+            }
+
+            _conversionInProgress = false;
+        }
+
+        private async Task SetInputValueAsync(string value)
+        {
+            await ThreadHelper.RunOnUIThreadAsync(() =>
+            {
+                _ignoreInputValueChange = true;
+                InputValue = value;
+                _ignoreInputValueChange = false;
+            });
+        }
+
+        private async Task<string> EncodeHtmlDataAsync(string? data)
+        {
+            if (string.IsNullOrWhiteSpace(data))
+            {
+                return string.Empty;
+            }
+
+            await TaskScheduler.Default;
+
+            string? encoded;
+            try
+            {
+                encoded = HttpUtility.HtmlEncode(data);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogFault("Html - Encoder", ex);
+                return ex.Message;
+            }
+
+            return encoded;
+        }
+
+        private async Task<string> DecodeCertificateDataAsync(string data, string? password)
+        {
+            if (string.IsNullOrWhiteSpace(data))
+            {
+                return string.Empty;
+            }
+
+            await TaskScheduler.Default;
+            string? decoded = string.Empty;
+            if (CertificateHelper.TryDecodeCertificate(data, password, out string? decodedValue))
+            {
+                decoded = decodedValue!;
+            }
+
+            return decoded;
+        }
+
+        private async Task SetCertificateDataAsync(StorageFile? file)
+        {
+            await ThreadHelper.RunOnUIThreadAsync(() =>
```

Comment:
Aren't we already on the UI thread when we invoke this method on line 146?

Reply:
Wow nice catch.  Will adjust, thanks.